app.py

```diff
@@ -1,12 +1,26 @@
 import os
 import time
 import gradio as gr
+import requests
 from dotenv import load_dotenv
 from griptape.chat_demo import Chat
+from griptape.chat_cloud import Chat_Cloud, Chat_Local
 
+# Load the environment variables
 load_dotenv()
+
+# Get the port from the environment variables for Gradio (default is 7860)
 port = os.getenv("GRADIO_PORT", 7860)
-chat = Chat()
+
+# Get the lambda endpoint from environment variables if using CDK based memory
+lambda_endpoint = os.getenv("LAMBDA_ENDPOINT", "")
+ 
+# Function to get the session id from the lambda endpoint
+def get_state():
+    resp = requests.post(lambda_endpoint, json={"operation": "create_session"})
+    session_id = resp.json()["session_id"]
+    return session_id
+
 
 def user(user_message, history):
     return "", history + [[user_message, None]]
```

Comment:
function name doesn't tell me what this does exactly. And it's doing some Python gymnastics